snapcraft/plugins/catkin.py

```diff
@@ -127,77 +160,59 @@ def gcc_version(self):
     @property
     def rosdir(self):
         return os.path.join(self.installdir,
-                            'opt', 'ros', self.options.rosversion)
+                            'opt', 'ros', self.options.rosdistro)
 
-    def _deps_from_packagesxml(self, f, pkg):
-        tree = lxml.etree.parse(f)
+    def _find_dependencies(self):
+        if self.package_deps_found:
+            return
 
-        for deptype in ('buildtool_depend', 'build_depend', 'run_depend'):
-            for xmldep in tree.xpath('/package/' + deptype):
-                dep = xmldep.text
+        for package in self.packages:
+            self._find_package_dependencies(package)
 
-                self.dependencies.append(dep)
+        self.package_deps_found = True
 
-                # Make sure we're not providing the dep ourselves
-                if dep in self.packages:
-                    self.package_local_deps[pkg].add(dep)
-                    continue
+    def _find_package_dependencies(self, package):
+        if package not in self.package_local_deps:
+            self.package_local_deps[package] = set()
 
-                # If we're already getting this through a deb package,
-                # we don't need it
-                if (dep in self._deb_packages or
-                        dep.replace('_', '-') in self._deb_packages):
-                    continue
+        # Query rosdep for the list of dependencies for this package
+        dependencies = self._rosdep.get_dependencies(package)
 
-                # Get the ROS package for it
-                self._deb_packages.append(
-                    'ros-'+self.options.rosversion+'-'+dep.replace('_', '-'))
+        for dependency in dependencies:
+            if not dependency:
```

Comment:
Yeah you're right, this should be checked in `get_dependencies()`. And to answer your question, it shouldn't, but since I'm parsing stdout I'm trying to account for small changes in the output of `rosdep`.